src/optimize.ts

```diff
@@ -277,35 +303,68 @@ async function processImage(
    */
   img.attr('decoding', 'async');
 
+  let buffer: Buffer | undefined;
+  let cdnTransformer: UrlTransformer | undefined;
+  let absolutePath: string | undefined; // only set for local attrib_src
+
   /*
    * Check for external images
    */
   if (!isLocal(attrib_src)) {
-    switch (config.image.external.process) {
-      case 'off': // Don't process external images
-        return;
-      case 'download': // Download external image for local processing
-        try {
-          attrib_src = await downloadExternalImage(htmlfile, attrib_src);
-          img.attr('src', attrib_src);
-        } catch (e: any) {
-          $state.reportIssue(htmlfile, {
-            type: 'warn',
-            msg: `Failed to download image: ${attrib_src} - ${e.message}`,
-          });
-          return; // No more processing
-        }
+    switch (config.image.cdn.process) {
+      case 'off':
         break;
+      case 'optimize':
+        const canonical = getCanonicalCdnForUrl(attrib_src);
+        if (!canonical) break;
+        cdnTransformer = getTransformer(canonical.cdn);
+        // need to download the image to get meta (size + transparency)
+        ({ buffer } = await downloadExternalImage(attrib_src));
```

Comment:
It could fail to download image here

Reply:
Should we start with an option where the file is not downloaded? It would be faster and less subject to network/download issues. We have the image size, we could just forget about transparency (and progressive jpeg) to begin with.

Reply:
We would have an option `optimize-without-download` and this would only process cdn images that have `width` and `height` - our use case

Reply:
FIP could add the "transparency" information as a `data-has-transparency` attribute to unlock `progressive-jpeg`.

Reply:
This would dramatically speedup the processing time in our usecase

Reply:
I see what you mean, but we don't really have the image size as the one we have is the one on the canvas, not the "full image size" the designer may have uploaded

We'd need to add them as data attributes too

Reply:
Note: If we go `srcset` with `2x` we don't need the size of the original image.

Reply:
> I see what you mean, but we don't really have the image size as the one we have is the one on the canvas, not the "full image size" the designer may have uploaded
> 
> We'd need to add them as data attributes too

hmmm getSizeAsync gives the original image size afaik https://www.figma.com/plugin-docs/api/Image/

Reply:
It does but we do need to provide it in additional attributes?

Reply:
nvm my ramblings, we can specify the full image size in standard width/height tags, that's actually what we should do